### PR TITLE
Add support for GL_EXT_shader_tile_image

### PR DESCRIPTION
**Pull request by janharaldfredriksen-arm**
Adds glslang support for GL_EXT_shader_tile_image - see https://github.com/KhronosGroup/GLSL/pull/203

### CODE REVIEW

**Comment by janharaldfredriksen-arm** [comment]
I had one issue when regenerating the grammar.

In glslang/MachineIndependent/glslang_tab.cpp, the first token got generated as "$end" rather than "end of file". This causes a couple of tests to fail in regression because the expected error messages assume "end of file".

I've manually modified the file like so:
```
-  "$end", "error", "$undefined", "CONST", "BOOL", "INT", "UINT", "FLOAT",
+  "\"end of file\"", "error", "$undefined", "CONST", "BOOL", "INT", "UINT", "FLOAT",
```
But obviously that's not great... Other suggestions welcome.

(The version of Bison I had was slightly older than what was used before, but I don't know if that's got anything to do with it. At least I couldn't find anything obvious in documentation.)

**Comment by arcady-lunarg** [comment]
The version of bison is in fact exactly the issue here, there was a change in 3.6 that changed the cryptically $end token to "end of file". If you could try regenerating the file using a newer bison, ideally 3.7.4 or later, it should work properly and is a better idea than tweaking the file by hand.

**Comment by janharaldfredriksen-arm** [comment]
Worked better with bison 3.7.4.

Any idea what's up with the Windows CI?
Log says:
>ERROR: C:/users/containeradministrator/_bazel_containeradministrator/rtpovv7a/external/com_google_googletest/BUILD.bazel:57:1: C++ compilation of rule '@com_google_googletest//:gtest' failed (Exit 1)

**Comment by arcady-lunarg** [comment]
The windows CI issue is a known problem due the required Visual Studio version being increased, I wouldn't worry about it. I'll take a look at your changes today.

**Comment by arcady-lunarg** [comment]
I'm still reading through the code, but in the meantime one piece of feedback. I want to thank you for writing tests, but one think I think would be good to have in addition to the negative tests for the error cases is a positive test that actually generates some SPIR-V.

**Review by arcady-lunarg — CHANGES_REQUESTED** [review]
Looks good but I have some questions about ABI issues with new enum variants, and I would like to see a test that generates actually generates some SPIR-V using the extension.

File: glslang/MachineIndependent/Initialize.cpp [hunk 1]
```diff
@@ -6219,7 +6241,6 @@ void TBuiltIns::add2ndGenerationSamplingImaging(int version, EProfile profile, c
                             continue;
                         if (ms && arrayed && profile == EEsProfile && version < 310)
                             continue;
-#endif
```
[the remark this answers is not in the document]
Comment (janharaldfredriksen-arm) [reply glslang/MachineIndependent/Initialize.cpp]: Good catch. A rebase bug, I think.

**Comment by janharaldfredriksen-arm** [comment]
> Looks good but I have some questions about ABI issues with new enum variants, and I would like to see a test that generates actually generates some SPIR-V using the extension.

Added a couple of positive tests in the latest commit.

**Review by arcady-lunarg — COMMENTED** [review]
LGTM, but the PR needs to be rebased because another PR touching the enums got merged first.

**Review by arcady-lunarg — APPROVED** [review]
Thanks, LGTM, I'll merge it as soon as the CI finishes.

**Comment by arcady-lunarg** [comment]
Turns out this was not in fact fine, you access `TIntermediate::usedIo[4]` when said array only has size 4 so it clobbers whatever follows it in the struct. And despite being enabled a while ago, ASAN didn't catch this bug until someone changed the size of `TIoRange`.

**Comment by arcady-lunarg** [comment]
> Turns out this was not in fact fine, you access `TIntermediate::usedIo[4]` when said array only has size 4 so it clobbers whatever follows it in the struct. And despite being enabled a while ago, ASAN didn't catch this bug until someone changed the size of `TIoRange`. Goes to show that I need to review much more carefully in the future.

**Pull request merged**